File: FirServer/FirServer/Manager/HandlerManager.cs
using FirServer.Define;
using FirServer.Interface;
using Google.Protobuf;
using log4net;

namespace FirServer.Manager
{
    public class HandlerManager : BaseManager
    {
        private static readonly ILog logger = LogManager.GetLogger(AppConst.LogRepos?.Name, typeof(HandlerManager));
        private Dictionary<string, IHandler> mHandlers = new Dictionary<string, IHandler>();

        public override void Initialize()
        {
        }

        /// <summary>
        /// 添加处理器
        /// </summary>
        public void AddHandler(string protocal, IHandler handler)
        {
            if (mHandlers.ContainsKey(protocal))
            {
                return;
            }
            mHandlers.Add(protocal, handler);
        }

        /// <summary>
        /// 添加处理器
        /// </summary>
        public IHandler? GetHandler(string protocal)
        {
            if (!mHandlers.ContainsKey(protocal))
            {
                return null;
            }
            return mHandlers[protocal];
        }

        /// <summary>
        /// 移除处理器
        /// </summary>
        public void RemoveHandler(string protocal)
        {
            mHandlers.Remove(protocal);
        }

        /// <summary>
        /// 接收数据
        /// </summary>
        internal async Task OnRecvData(string protoName, MsgChannel channel, ByteString byteData)
        {
            mHandlers.TryGetValue(protoName, out IHandler? handler);
            if (handler != null)
            {
                try
                {
                    await handler.OnMessage(channel, byteData);
                }
                catch (Exception ex)
                {
                    logger.Error(ex.Message);
                }
            }
            else
            {
                logger.Error("Proto [" + protoName + "] not found!~~");
            }
        }

        public override void OnDispose()
        {
        }
    }
}

File: FirServer/FirServer/src/com/manager/HandlerManager.java
package com.manager;

import java.util.LinkedHashMap;
import java.util.Map;

import com.interfaces.IHandler;
import com.smartfoxserver.v2.entities.User;

public class HandlerManager extends BaseManager {
    private static Map<String, IHandler> handlers = new LinkedHashMap<String, IHandler>();

    @Override
    public void Initialize() {
    }

    public void AddHandler(String typeName, IHandler handler) {
        if (!handlers.containsKey(typeName)) {
            handlers.put(typeName, handler);
        }
    }

    public IHandler GetHandler(String typeName) {
        IHandler handler = null;
        if (handlers.containsKey(typeName))
        {
            handler = handlers.get(typeName);
        }
        return handler;
    }

    public void RemoveHandler(String typeName) {
        if (handlers.containsKey(typeName)) {
            handlers.remove(typeName);
        }
    }

    public void OnRecvMessage(User user, String protoName, byte[] bytes) {
        IHandler handler = GetHandler(protoName);
        if (handler != null) {
            handler.OnMessage(user, bytes);
        }
    }

    @Override
    public void OnDispose() {
    }


}
